### ISSUE TITLE
Monitor without variable names, or fail gracefully

### ISSUE THREAD

**Issue by bartvm** [bartvm/blocks#160]
I've made this mistake several times now: When e.g. adding regularization manually, the variable doesn't have a name, resulting in:
```
      File "/home/vanmerb/lisa/blocks-master/blocks/extensions/monitoring.py", line 15, in _add_records
        prefixed_name = prefix + PREFIX_SEPARATOR + name
    TypeError: cannot concatenate 'str' and 'NoneType' objects
```
Would be nice to give an informative error, or maybe we can use `str(var)` or do something else intelligently.

**Comment by rizar** [comment]
I suggest we just fail gracefully, because there is no good from monitoring something called `Elemwise.0`.

**Comment by bartvm** [comment]
Fixed in 0a889f4537ad7992eac5cbd0655fe9b991293325
